csm_web/frontend/src/components/course/ManageLabelsModal.tsx

```diff
@@ -0,0 +1,207 @@
+import React, { useEffect, useState } from "react";
+import { fetchJSON, fetchWithMethod, HTTP_METHODS } from "../../utils/api";
+import LoadingSpinner from "../LoadingSpinner";
+import Modal from "../Modal";
+import { Label } from "../../utils/types";
+import Pencil from "../../../static/frontend/img/pencil.svg";
+import Trash from "../../../static/frontend/img/trash-alt.svg";
+import Exit from "../../../static/frontend/img/check_circle.svg";
+
+interface ManageLabelsModalProps {
+  closeModal: () => void;
+  courseId: number;
+  reloadSections: () => void;
+  title: string | boolean;
+}
+
+/**
+ * Modal that coords use to export a csv of student emails in selected courses.
+ */
+export const ManageLabelsModal = ({
+  closeModal,
+  courseId,
+  reloadSections,
+  title
+}: ManageLabelsModalProps): React.ReactElement => {
+  /**
+   * Map of course id to course name.
+   */
+  const [courseLabels, setCourseLabels] = useState<Label[]>([]);
+
+  const [currCourseLabels, setCurrCourseLabels] = useState<Label[]>([]);
+
+  const [newLabelKey, setNewLabelKey] = useState(-1);
```

Comment:
I don't believe the first docstring is accurate? I'd also recommend adding docstrings for the rest of the states.

Reply:
donsies !